### ISSUE TITLE
[Kiota] CreateZipArchive = false is not working

### ISSUE THREAD

**Issue by zaksnet** [FastEndpoints/FastEndpoints#585]
CreateZipArchive = false does not seem to have an effect. It still generates a zip file

**Comment by dj-nitehawk** [comment]
if you're doing `.MapApiClientEndpoint()` then `CreateZipArchive` property has no effect because a zip archive **must** be generated in order to provide a downloadable byte stream.

so to reiterate, `CreateZipArchive` is only effective for `.GenerateApiClientsAndExitAsync()`

can you confirm which method you're using? if it's the latter, then it's a bug we need to look in to.

thanks!

**Comment by zaksnet** [comment]
I see, i am using `.MapApiClientEndpoint()`  and now that i'v played around with it, it makes sense that it creates a zip because the generated client consists of multiple files. Thanks 👌

**Comment by dj-nitehawk** [comment]
it is however supposed to delete the generated zip file after the response has completed.

https://github.com/FastEndpoints/FastEndpoints/blob/305921b0b05c90b1872ec232a82c1847678b9386/Src/ClientGen.Kiota/Extensions.cs#L41-L49

maybe you were seeing a zipfile from a broken/interrupted response?

**Comment by zaksnet** [comment]
I did not provide an OutputPath, as I was only interested in the response content. When I tried to debug the code, I found that the file was being stored in `c.ZipOutputFile!`, which is `ClientGen\..\FrontendConsumersApiClient.zip`. This leads to zipFile being saved in the root directory of my project and then properly deleted.

I am confused about the purpose of `Directory.Delete(c.OutputPath, true);`. It seems risky because if someone enters the wrong path, they could end up losing files. I would assume that all we need to do is make sure that the directory exists
